packages/ses/src/error/tame-v8-error-constructor.js

```diff
@@ -299,8 +301,37 @@ export const tameV8ErrorConstructor = (
     // Use concise methods to obtain named functions without constructors.
     const systemMethods = {
       prepareStackTrace(error, sst) {
-        weakmapSet(stackInfos, error, { callSites: sst });
-        return inputPrepareFn(error, safeV8SST(sst));
+        const stackInfo = {};
+        const safeCallSites = safeV8SST(sst);
+        let preparedStack;
+        try {
+          preparedStack = inputPrepareFn(error, safeCallSites);
+          if (preparedStack === safeCallSites) {
+            // Handle depd and similar prepareStackTrace which return the structured stack trace
+            stackInfo.callSites = sst;
+            return safeCallSites;
+          } else if (
+            typeof preparedStack === 'string' &&
+            preparedStack !== ''
+          ) {
+            stackInfo.stackString = stringReplace(preparedStack, /^[\n]+/, '');
```

Comment:
This stringReplace does not seem to correspond to anything in the original. Is it an observable behavior change? What is its purpose?

It seems to only remove an initial run of newlines and not do anything else. If this is a behavioral change, it is a good one! When does this happen? (Seem to ring a bell, but I don't remember)

Reply:
a `stack` property is of the form `ErrorName: error message(\n  at location:12:3)+`.
`prepareStack` is supposed to return a full `stack` property.

`stackString` seems to be only the formatted location lines, missing the error name and message. This reconciliates the 2 formats.

This behavior wasn't in the original code because the taming never used the return value of the user specified `prepareStack` to compute its own stack info.